app/controllers/posts_controller.rb

```diff
@@ -1,18 +1,20 @@
 class PostsController < ApplicationController
   before_action :logged_in_user, only: [:create]
 
-  def show
-    @post = Post.find(params[:id])
-  end
-
   def new
     @post = Post.new
   end
 
+  def show
+    @post = Post.find(params[:id])
+    @comments = @post.comments.order('created_at ASC')
+    @comment = Comment.new
```

Comment:
I'm not trying to create new comments related to `@post`, but rather the `@comments` variable is simply pulling out all comments associated with a given post (a post `has_many` comments and a comment `belongs_to` a post.) Sorry, if I misunderstand your comment.